pkg/collector/corechecks/cluster/orchestrator/transformers/k8s/common_test.go

```diff
@@ -23,3 +29,48 @@ func int64Ptr(v int64) *int64 {
 func strPtr(v string) *string {
 	return &v
 }
+
+func getTemplateWithResourceRequirements() corev1.PodTemplateSpec {
+	parseRequests := resource.MustParse("250M")
+	parseLimits := resource.MustParse("550M")
+	return corev1.PodTemplateSpec{
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				{
+					Name: "aContainer",
+					Resources: corev1.ResourceRequirements{
+						Limits:   map[corev1.ResourceName]resource.Quantity{corev1.ResourceMemory: parseLimits},
+						Requests: map[corev1.ResourceName]resource.Quantity{corev1.ResourceMemory: parseRequests},
+					},
+				},
+			},
+			InitContainers: []corev1.Container{
+				{
+					Name: "aContainer",
+					Resources: corev1.ResourceRequirements{
+						Limits:   map[corev1.ResourceName]resource.Quantity{corev1.ResourceMemory: parseLimits},
+						Requests: map[corev1.ResourceName]resource.Quantity{corev1.ResourceMemory: parseRequests},
+					},
+				},
+			},
+		},
+	}
+}
+
+func getExpectedModelResourceRequirements() []*model.ResourceRequirements {
+	parseRequests := resource.MustParse("250M")
+	parseLimits := resource.MustParse("550M")
+	return []*model.ResourceRequirements{
+		{
+			Limits:   map[string]int64{corev1.ResourceMemory.String(): parseLimits.Value()},
+			Requests: map[string]int64{corev1.ResourceMemory.String(): parseRequests.Value()},
+			Name:     "aContainer",
+			Type:     model.ResourceRequirementsType_container,
+		}, {
+			Limits:   map[string]int64{corev1.ResourceMemory.String(): parseLimits.Value()},
+			Requests: map[string]int64{corev1.ResourceMemory.String(): parseRequests.Value()},
+			Name:     "aContainer",
+			Type:     model.ResourceRequirementsType_initContainer,
+		},
+	}
+}
```

Comment:
it's just my feeling but I would have moved to the pod code (since the template is a pod spec) because it makes sense for some resources to depend on the pod transformer (hence depend on its test tooling too)

conversely, not all resources might have a pod spec

Reply:
I moved this to `pod_test.go` let me know if you meant otherwise